### ISSUE TITLE
Refactor losses to inherit from base keras.Loss

### ISSUE THREAD

**Issue by seanpmorgan** [tensorflow/addons#259]
Per recommendation in https://github.com/tensorflow/tensorflow/issues/28601 we should aim to inherit from base `Loss`. I believe this only requires a minor refactoring so replicate the results of `LossWrapper`
